### DISCUSSION TITLE
an error occurred while fetching api/client-anr/2/instances-risks/

### DISCUSSION THREAD

**Discussion by CarlosAugustoDeSouzaJunior** [monarc-project/MonarcAppFO#488] [General]
This bug tracker is for tracking bugs, todos and confirmed new features.
If you want to submit a feature request please discuss about it here:
https://github.com/monarc-project/MonarcAppFO/discussions

**Describe the bug**
Cannot change the prob or Qualif, it is generating the error. No error in the install process.

**To Reproduce**
1. Go to risk analyses, change the impat value, in the treath colum, change the prob to any number. Same thing when add Qualif.
2. See error4
" an error occurred while fetching api/client-anr/2/instances-risks/98: class "Locale" not found (500)"

**Expected behavior**
Be able to change the prob et Qualif without error.

**Screenshots**

![image](https://github.com/monarc-project/MonarcAppFO/assets/34326773/208be71a-7856-457c-9d12-65740142ac14)

![image](https://github.com/monarc-project/MonarcAppFO/assets/34326773/f02d3d7c-95e3-48cf-9512-76eec648bcf8)

**Desktop (please complete the following information):**

| Questions                                | Answers
|-----------------------------------------|--------------------
| Type of issue                           | Bug
| OS version (server)                 |  Ubuntu, 22.04.
| OS version (client)                   | Win10
| PHP version                             | 8.1
| MONARC version / git hash     | 2.12.6
| Browser                                    | Chrome

**Additional context**
Only front end installe.

**Comment by ruslanbaidan** [comment]
We can't reproduce the error on our side and the error message that you mentioned "class "Locale" not found (500)" , could only mean that some php external dependencies were not installed properly.
You could try to remove the `vendor` directory from the project root:
`rm -rf vendor`
and try to reinstall all the dependencies:
`composer ins --no-cache --ignore-platform-req=php`

I'm going to convert the issue to discussion as there is no issue, but installation or configuration related challenge.

**Reply by CarlosAugustoDeSouzaJunior** [reply]
I try it, but broke everithing. and I thing that compose ins not in insatall intruction for ubuntu 20.04, I had installed it and ran the command,  got an erro about the ignore, now nothing is working:)

![image](https://github.com/monarc-project/MonarcAppFO/assets/34326773/da4c1c5d-014b-4b41-ad36-a0ea50a5831f)

**Reply by CarlosAugustoDeSouzaJunior** [reply]
Hi Ruslan,

I just create a brand new (Azure) ubuntu 2204, ran the script bootstrap.sh (packer), it installed everything without error. but now I cannot even an anlises.

![image](https://github.com/monarc-project/MonarcAppFO/assets/34326773/ff4ae81b-b52a-473d-a08e-a599f79dcbaa)

![image](https://github.com/monarc-project/MonarcAppFO/assets/34326773/575282e5-5b5b-4ad9-95a2-bbd115784c55)

HELP!!

Nothing is listed anywhre...

**Reply by ruslanbaidan** [reply]
> I try it, but broke everithing. and I thing that compose ins not in insatall intruction for ubuntu 20.04, I had installed it and ran the command, got an erro about the ignore, now nothing is working:)
> 
> ![image](https://user-images.githubusercontent.com/34326773/241831862-da4c1c5d-014b-4b41-ad36-a0ea50a5831f.png)

1. `composer ins --no-cache --ignore-platform-req=php` has to be run from the project's root directory, as it says that `composer.json` is not found and it's in the root. There is also an option `--ignore-platform-reqs` to ignore all the requirements of composer.json.
It also looks like your composer is outdated, so better to update it before run.

2. You don't need to bootstrap it again, perhaps due to missing composer installations ./scripts/update-all.sh -d was not run properly that could lead to missing initial data and migrations.

When the composer command is executed without any errors, it would be possible to run:
`./scripts/update-all.sh` or `./scripts/upgrade-db.sh`

**Comment by CarlosAugustoDeSouzaJunior** [comment]
I found the problem, I have ajusted the Pakcer Script to work in a AzureVM, so the ... sudo -u monarc..cause le probleme. Thank you. I removed using now ...sudo mysql -u$......

**Reply by Coderyang666666** [reply]
I can't even locate this api file under wsl, I also have a similar error, can you help me solve it, thanks!
<img width="256" alt="image" src="https://github.com/monarc-project/MonarcAppFO/assets/90095020/b9ee083b-7cf4-4605-97a1-fbf520fd828b">
